OpenTrons/opentrons-api
# feature(api, hardware-controller): allow hardware to move both mounts at the same time

# Overview

Move both mounts for 96c attach/detach flow.
`move_axes` in hardware controller.
Allow `machine_from_deck` to use more than 3 axis. 
Use point instead of slot for `move_to_maintenance_position` https://opentrons.atlassian.net/browse/RSS-244

# Test Plan

- Use the app on at OT-3 to attach/detach a 96c. make sure both mounts come down the z to the same location and allow easy attaching.
- Use app to test attach/detach flows with 1-8c. make sure works as expected. 
- Test protocols with movement commands and make sure it's all working as expected. we made a change to the coordinates machine calculations and we need to make sure nothing has changed. 

# Changelog

- Changed `move_to_maintenance_position`:

1. use 2 modes - AttachPlate for 96c and AttachInstrument for all other instruments.
2. make 2 calls to move_axes for both mounts when using 96c. 
3. use point instead of slot with offset for the attach/detach position. 

- added hardware `move_axes` to move a list of axis instead of a mount that move_to uses. 
- refactored `machine_from_deck` to accept more than 3 axes (to be able to move both mounts at the same time - currently not supported by head firmware)


# Review requests

does changes make sense? did I forget anything?

# Risk assessment

Medium. changes how we calculate `machine_from_deck` and this should affect all movement commands. `move_axes` should only affect `move_to_maintenance_poisition` but need to make sure nothing in the attach/detach flows is broken.
